### PR TITLE
Feature/manage stores

### PR DESCRIPTION
**Pull request by simaob**
In this PR:

- Adds UserStores model to allow linking store managers to stores;
- Updates user edit page to enable associating stores with managers;
- Adds a new route to view the current user stores;
- Editing the status of the store by the manager;

ToDo:
- Review all authorization to ensure we are good!

### CODE REVIEW

File: app/controllers/stores_controller.rb [hunk 1]
```diff
@@ -12,7 +12,16 @@ def index
...
     end
 
-    @stores = @stores.order(:group, :name).page(params[:page])
+    @stores = @stores.order(:group, :name)
+
+    respond_to do |format|
+      format.html do
+        @stores = @stores.page(params[:page])
+      end
+      format.json do
```
Comment (santostiago) [inline app/controllers/stores_controller.rb]: We won't really need this, as the json requests are handled by the api...

File: app/views/stores/index.json.jbuilder [hunk 2]
```diff
@@ -1 +1,3 @@
-json.array! @stores, partial: "stores/store", as: :store
+json.results do
+  json.array! @stores, partial: "stores/store", as: :store
```
Comment (santostiago) [inline app/views/stores/index.json.jbuilder]: Do we need this?

Comment (simaob) [reply app/views/stores/index.json.jbuilder]: Yes. It's used by the select2 search

**Comment by simaob** [comment]
I've added screens to manage stores' status

Screenshot 2020-03-29 at 17 04 45
Screenshot 2020-03-29 at 17 04 50

**Comment by simaob** [comment]
rebased

File: app/models/status_store_owner.rb [hunk 3]
```diff
@@ -20,5 +20,12 @@
 #  active                :boolean          default("true")
 #
 class StatusStoreOwner < Status
+  after_validation :set_valid_until
+  validates :updated_time, :status, :store_id, presence: true
 
+  private
+
+  def set_valid_until
+    self.valid_until = self.updated_time + 1.hour
```
Comment (santostiago) [inline app/models/status_store_owner.rb]: This could be a constant...

File: app/models/store.rb [hunk 4]
```diff
@@ -52,10 +54,18 @@ def address
...
+  def text
+    [name, city].compact.join(', ')
+  end
+
   def self.groups
     select(:group).order(:group).distinct.pluck(:group)
   end
 
+  def latest_owner_status
+    status_store_owners.order(updated_at: :desc).limit(1)&.first
```
Comment (santostiago) [inline app/models/store.rb]: Shouldn't we have an "active" flag for this? (it might be in my PR)

**Pull request merged**